### PR TITLE
[ticket/16912] Improve mail encoding to better match RFC 2047

### PR DESCRIPTION
**Pull request by Neo-CTC**
Checklist:

- [ ] Correct branch: master for new features; 3.3.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [ ] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

[PHPBB3-16912](https://tracker.phpbb.com/browse/PHPBB3-16912)

### CODE REVIEW

**Comment by 3D-I** [comment]
Please see: https://github.com/phpbb/phpbb/pull/6320 - as I suggested in the tracker.

File: phpBB/includes/functions_messenger.php [hunk 1]
```diff
@@ -1853,53 +1853,16 @@ function digest_md5($username, $password)
...
-
-		while (count($array) && intval((strlen($text . $array[0]) + 2) / 3) << 2 <= $split_length)
-		{
-			$text .= array_shift($array);
-		}
-
-		$str .= $start . base64_encode($text) . $end . $delimiter;
-	}
+	// Is it more efficient to use quoted or base64 encoding?
+	$transfer = strlen($str) / utf8_strlen($str) < 1.4 ? 'Q' : 'B';
```
Comment (rxu) [inline phpBB/includes/functions_messenger.php:1863]: Could you elaborate `1.4` choice?

File: phpBB/includes/functions_messenger.php [hunk 2]
```diff
@@ -1853,53 +1853,16 @@ function digest_md5($username, $password)
...
-		return $start . $encoded_str . $end;
-	}
-
-	// If there is only ASCII data, we just return what we want, correctly splitting the lines.
-	if ($is_ascii)
-	{
-		return $start . implode($end . $delimiter . $start, str_split($encoded_str, $split_length)) . $end;
-	}
+	// Correctly decode html entities
+	$str = html_entity_decode($str);
```
Comment (3D-I) [inline phpBB/includes/functions_messenger.php:1857]: Looks like something new awaiting to be approved.

https://github.com/phpbb/phpbb/pull/6320/files#diff-def4c3d80a12a0d27ddce789bcc0809ead44c8e488e2f05181d8846e1a0e5324R1866-R1867

File: phpBB/includes/functions_messenger.php [hunk 3]
```diff
@@ -1853,53 +1853,16 @@ function digest_md5($username, $password)
...
-			$text .= array_shift($array);
-		}
-
-		$str .= $start . base64_encode($text) . $end . $delimiter;
-	}
+	// Is it more efficient to use quoted or base64 encoding?
+	$transfer = strlen($str) / utf8_strlen($str) < 1.4 ? 'Q' : 'B';
 
-	return substr($str, 0, -strlen($delimiter));
+	return mb_encode_mimeheader($str, $encoding, $transfer, $eol);
```
Comment (rxu) [inline phpBB/includes/functions_messenger.php:1865]: For the note https://github.com/symfony/polyfill-mbstring/blob/344e456152e22a1bce3048c6c311059ea14bde47/Mbstring.php#L138
Using `iconv_mime_encode()` will require Iconv extension to be a mandatory for phpBB though.

**Comment by marc1706** [comment]
Thank you for your pull request. After careful consideration, we have however opted for the solution provided by https://github.com/phpbb/phpbb/pull/6332

I will therefore close this.

**Pull request closed without merging**